tests/test_boolean.py

```diff
@@ -324,217 +195,419 @@ def test_parse_side_by_side_symbols_with_parens_raise_exception(self):
+        assert algebra.NOT(s).isliteral
+        # negation of a non-literal is still a non-literal
+        assert not algebra.parse('~(a|b)').isliteral
+
+    def test_literals_0(self):
         algebra = BooleanAlgebra()
+
         a = algebra.Symbol('a')
-        l = ~a
-        self.assertTrue(l.isliteral)
-        self.assertTrue(l in l.literals)
-        self.assertEqual(len(l.literals), 1)
+        b = ~a
+
+        assert a.isliteral
+        assert b.isliteral
+
+        assert a in a.literals
+        assert b in b.literals
+
+        assert len(a.literals) == 1
+        assert len(b.literals) == 1
+
+    def test_literals_1(self):
+        algebra = BooleanAlgebra()
 
-        l = algebra.parse('~(a&a)')
-        self.assertFalse(l.isliteral)
-        self.assertTrue(a in l.literals)
-        self.assertEqual(len(l.literals), 1)
+        expression = algebra.parse('~(a&a)')
 
-        l = algebra.parse('~(a&a)', simplify=True)
-        self.assertTrue(l.isliteral)
+        assert not expression.isliteral
+
+        assert algebra.Symbol('a') in expression.literals
+        assert len(expression.literals) == 1
+
+    def test_literals_2(self):
+        algebra = BooleanAlgebra()
+
+        expression = algebra.parse('~(a&a)', simplify=True)
+
+        assert expression.isliteral
+        assert expression == algebra.NOT(algebra.Symbol('a'))
 
     def test_literalize(self):
         parse = BooleanAlgebra().parse
-        self.assertEqual(parse('~a').literalize(), parse('~a'))
-        self.assertEqual(parse('~(a&b)').literalize(), parse('~a|~b'))
-        self.assertEqual(parse('~(a|b)').literalize(), parse('~a&~b'))
 
-    def test_simplify(self):
+        assert parse('~a').literalize() == parse('~a')
+        assert parse('~(a&b)').literalize() ==  parse('~a|~b')
+        assert parse('~(a|b)').literalize() ==  parse('~a&~b')
+
+    def test_invert_eq_not(self):
         algebra = BooleanAlgebra()
+
         a = algebra.Symbol('a')
-        self.assertEqual(~a, ~a)
-        assert algebra.Symbol('a') == algebra.Symbol('a')
-        self.assertNotEqual(a, algebra.parse('~~a'))
-        self.assertEqual(a, (~~a).simplify())
-        self.assertEqual(~a, (~~ ~a).simplify())
-        self.assertEqual(a, (~~ ~~a).simplify())
-        self.assertEqual((~(a & a & a)).simplify(), (~(a & a & a)).simplify())
-        self.assertEqual(a, algebra.parse('~~a', simplify=True))
-
-    def test_cancel(self):
+
+        assert ~a == ~a
+        assert ~a == algebra.NOT(a)
+
+    def test_simplify(self):
         algebra = BooleanAlgebra()
+
         a = algebra.Symbol('a')
-        self.assertEqual(~a, (~a).cancel())
-        self.assertEqual(a, algebra.parse('~~a').cancel())
-        self.assertEqual(~a, algebra.parse('~~~a').cancel())
-        self.assertEqual(a, algebra.parse('~~~~a').cancel())
 
-    def test_demorgan(self):
+        assert a == a.simplify()
+        assert a == (~~a).simplify()
+        assert a == (~~ ~~a).simplify()
```

Comment:
I think the space  in the middle of the `~~ ~~` here and there is likely a typo that I introduced in the past from some editor reformatting.